Handle GitHub token revocation

e18r opened the pull request:
Fixes #654 

When a GitHub request is made with a revoked token, the end user is informed and asked to sign in again.

![Peek 2020-01-07 10-12](https://user-images.githubusercontent.com/16065447/71907851-e3ec4d80-313a-11ea-92a9-4e864bd54fda.gif)

e18r committed:
fix(projects): handle GitHub token revocation

When a GitHub request is made with a revoked token, the end user is
informed and asked to sign in again.

chadoh commented on apps/projects/app/utils/apollo-client.js:
```diff
@@ -1,16 +1,34 @@
-  })
+    })
+  }
+  return initApolloClient
```
I think we can save all consumers of `useApolloClient` a line or two by doing something like this:

```js
export const useApolloClient = () => {
  const { api, appState } = useAragonApi()
  if (appState.github.token === null) return null
  return new ApolloClient({ ... })
}
```

Note that there's also no reason to pass `token` in as an argument anymore, now that this is a hook!

Do you think this would work? Is there a reason to make this return an `initApolloClient` function, rather than `null` or an actual client?

chadoh commented on apps/projects/app/utils/apollo-client.js:
```diff
@@ -1,16 +1,34 @@
+            scope: null,
+            token: null,
+          })
+        }
```
Let's add an `else` with `console.error(error)`, to keep the functionality we had before and make sure these errors are still visible in the console

e18r committed:
fix(projects): simplified useApolloClient hook

Previously, in order to use the Github API client, you had to:

1. Get the GitHub token from the appState
2. Call the `useApolloClient` hook and get the `initApolloClient` function
3. Call the `initApolloClient` function with the GitHub token as an
argument and get the client

But since `useApolloClient` is a hook, this commit simplifies it so
that the appState is accessed directly from within the hook, and the
actual API client is returned directly, so that the above steps get
simplified to:

1. Call the `useApolloClient` hook and get the client

e18r commented:
Hey @chadoh, this PR is ready for another round of review. Thanks!

e18r commented:
Rebased with `dev` and solved several conflicts.

chadoh commented on apps/projects/app/components/Card/Revoked.js:
```diff
@@ -0,0 +1,40 @@
+import PropTypes from 'prop-types'
+import { Button, EmptyStateCard, GU, Text, useTheme } from '@aragon/ui'
+import { EmptyWrapper } from '../Shared'
+import revoked from '../../assets/revoked.svg'
```
Can we export a component, instead of raw SVG? Like we did for [`IconDoubleCheck` in Allocations](https://github.com/AutarkLabs/open-enterprise/pull/1853/files#diff-373c0e728dced2965861263f8f033827)

chadoh commented on apps/projects/app/components/Card/Revoked.js:
```diff
@@ -0,0 +1,40 @@
+
+const Illustration = () => <img src={revoked} alt="" />
+
+
```
our linter doesn't complain about two line breaks in a row so I guess I have to 🙃

chadoh commented on apps/projects/app/utils/apollo-client.js:
```diff
@@ -1,16 +1,35 @@
+      github: { token } = { token: null }
+    }
+  } = useAragonApi()
+  if (token === null) return null
```
This is all a bit weird to read. How about:

```js
const { api, appState: { github } } = useAragonApi()
if (!github || !github.token) return null
```

Then we'll have to use `github.token` elsewhere, but I think that's a good balance of readability here vs terseness there.

chadoh reviewed (changes requested):
This looks really nice! I see a couple more small improvements we could make, which would not require the "request changes" status, but I also see something else...

The IssueDetail page doesn't work! I haven't looked into why, but it works on `dev` and is broken on this branch.
